app/models/post.rb

```diff
@@ -2,4 +2,12 @@ class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
   has_many :comments
   has_many :likes
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
+
```

Comment:
- [ OPTIONAL] Amazing job for using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save in comment and like models  to call the #update_comments_counter & update_likes_counter methods, so I highly recommend adding it here as well that every time a new post has been saved to database, then make the `update_post_counter` method private 🎉🔥🚀

Reply:
Thank you @Reem-lab